contracts/proxy/transparent/ProxyAdmin.sol

```diff
@@ -12,6 +12,8 @@ import {Address} from "../../utils/Address.sol";
  * explanation of why you would want to use this see the documentation for {TransparentUpgradeableProxy}.
  */
 contract ProxyAdmin is Ownable {
+    uint256 public immutable PROXY_ADMIN_VERSION = 2;
```

Comment:
What about doing a string "5.0.0". It would be less than 32 bytes, so it would fit into a single slot ... so it might not be more expensive than the uint256 approach

Reply:
I would be in favor of this idea.